composer/datasets/cifar.py

```diff
@@ -49,10 +68,84 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
                 memory_format=self.synthetic_memory_format,
             )
 
+        elif self.use_ffcv:
+            try:
+                import ffcv  # type: ignore
+                from ffcv.fields.decoders import IntDecoder, SimpleRGBImageDecoder  # type: ignore
+                from ffcv.pipeline.operation import Operation  # type: ignore
+            except ImportError:
+                raise ImportError(
+                    textwrap.dedent("""\
+                    Composer was installed without ffcv support.
+                    To use ffcv with Composer, run `pip install mosaicml[ffcv]`"""))
+
+            dataset_file = self.ffcv_dest_train if self.is_train else self.ffcv_dest_val
+            dataset_filepath = os.path.join(self.ffcv_dir, dataset_file)
+            # always create if ffcv_write_dataset is true
+            if self.ffcv_write_dataset and not os.path.exists(dataset_filepath):
+                log.info(f"Writing dataset in FFCV <file>.beton format to {dataset_filepath}.")
+                if self.datadir is None:
+                    raise ValueError("datadir is required if use_synthetic is False and ffcv_write_dataset is True.")
+                ds = CIFAR10(
+                    self.datadir,
+                    train=self.is_train,
+                    download=self.download,
+                )
+
+                writer = ffcv.writer.DatasetWriter(dataset_filepath, {
+                    'image': ffcv.fields.RGBImageField(),
+                    'label': ffcv.fields.IntField()
+                })
+                writer.from_indexed_dataset(ds)
+
+            if not os.path.exists(dataset_filepath):
+                raise ValueError(
+                    f"Dataset file containing samples not found at {dataset_filepath}. Use ffcv_dir flag to point to a dir containing {dataset_filepath}."
+                )
+
+            # Please note that this mean/std is different from the mean/std used for regular PyTorch dataloader as
```

Comment:
oh man.....is there a way around this? ideally these constants would be the same regardless of dataset format.

Reply:
For CIFAR, it seems there isn't as the way these are used is different between torch vision and ffcv. However, the good news it's the same for ImageNet dataset.